# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/alessandro/busybox/lxconfig.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./miscutils/time.c.orig	2005-11-04 13:26:21.000000000 +0100
+++ ./miscutils/time.c	2005-11-04 13:26:28.000000000 +0100
@@ -21,6 +21,7 @@
    Heavily modified for busybox by Erik Andersen <andersen@codepoet.org>
    */
 
+#include "busybox.h"
 #include <stdlib.h>
 #include <stdio.h>
 #include <signal.h>
@@ -36,7 +37,6 @@
 
 #define TV_MSEC tv_usec / 1000
 #include <sys/resource.h>
-#include "busybox.h"
 
 /* Information on the resources used by a child process.  */
 typedef struct
--- ./shell/ash.c.orig	2005-11-04 13:27:12.000000000 +0100
+++ ./shell/ash.c	2005-11-04 13:27:21.000000000 +0100
@@ -65,6 +65,7 @@
 #define _GNU_SOURCE
 #endif
 
+#include "busybox.h"
 #include <sys/types.h>
 #include <sys/cdefs.h>
 #include <sys/ioctl.h>
@@ -96,7 +97,6 @@
 #include <fnmatch.h>
 
 
-#include "busybox.h"
 #include "pwd_.h"
 
 #ifdef CONFIG_ASH_JOB_CONTROL
--- ./sysklogd/syslogd.c.orig	2005-11-04 13:27:36.000000000 +0100
+++ ./sysklogd/syslogd.c	2005-11-04 13:27:43.000000000 +0100
@@ -26,6 +26,8 @@
  *
  */
 
+#include "busybox.h"
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <ctype.h>
@@ -45,8 +47,6 @@
 #include <sys/un.h>
 #include <sys/param.h>
 
-#include "busybox.h"
-
 /* SYSLOG_NAMES defined to pull some extra junk from syslog.h */
 #define SYSLOG_NAMES
 #include <sys/syslog.h>
--- ./networking/ping.c.orig	2005-11-04 13:26:37.000000000 +0100
+++ ./networking/ping.c	2005-11-04 13:26:45.000000000 +0100
@@ -31,6 +31,7 @@
  * Original copyright notice is retained at the end of this file.
  */
 
+#include "busybox.h"
 #include <sys/param.h>
 #include <sys/socket.h>
 #include <sys/file.h>
@@ -49,7 +50,6 @@
 #include <unistd.h>
 #include <string.h>
 #include <stdlib.h>
-#include "busybox.h"
 
 
 static const int DEFDATALEN = 56;
--- ./include/busybox.h.orig	2005-11-04 13:21:57.000000000 +0100
+++ ./include/busybox.h	2005-11-04 13:30:10.000000000 +0100
@@ -24,6 +24,13 @@
 #ifndef	_BB_INTERNAL_H_
 #define	_BB_INTERNAL_H_    1
 
+// never include the linux kernel config setting. They use the same
+// namespace as the busybox config flags.  -- Clifford Wolf
+#ifdef _LINUX_CONFIG_H
+#  error Linux kernel config included before busybox.h
+#endif
+#define _LINUX_CONFIG_H
+
 #include "config.h"
 
 #include <stdio.h>
